erpnext/stock/report/stock_balance/stock_balance.py

```diff
@@ -259,7 +259,9 @@ def apply_conditions(query, filters):
 	sle = frappe.qb.DocType("Stock Ledger Entry")
 	warehouse_table = frappe.qb.DocType("Warehouse")
 
-	if not filters.get("from_date"):
+	if from_date := filters.get("from_date"):
+		query = query.where(sle.posting_date >= from_date)
```

Comment:
This is incorrect. Stock balance report has to fetch all SLEs, the from date is only used for computing opening qty. 

This is somewhat inefficient but that's how it's correctly implemented.

Reply:
@s-aga-r

Reply:
also refer https://github.com/frappe/erpnext/issues/29183